File: backend/node/src/aws/LabelsRekognition.mjs
import { DetectLabelsCommand } from "@aws-sdk/client-rekognition";
import { RekognitionClient } from "@aws-sdk/client-rekognition";
import config from "../config.mjs";

// Esto es para reconocer etiquetas en una imagen
const rekogClient = new RekognitionClient({
  region: config.region,
  credentials: {
    accessKeyId: config.accessKeyId_s3,
    secretAccessKey: config.secretAccessKey_s3,
  },
});
const bucket = config.bucket_s3;

const detect_labels = async (photo) => {
  try {
    const params = {
      Image: {
        S3Object: {
          Bucket: bucket,
          Name: photo,
        },
      },
    };

    const response = await rekogClient.send(new DetectLabelsCommand(params));
    const etiquetas = [];
    response.Labels.forEach((label) => {
      etiquetas.push(label.Name);
      /*console.log(`Confidence: ${label.Confidence}`);
      console.log(`Name: ${label.Name}`);
      console.log("Instances:");
      label.Instances.forEach((instance) => {
        console.log(instance);
      });
      console.log("Parents:");
      label.Parents.forEach((name) => {
        console.log(name);
      });
      console.log("-------");*/
    });
    return etiquetas; // For unit tests.
  } catch (err) {
    console.log("Error", err);
    return null;
  }
};

export const rekognition = { detect_labels };

File: backend/python/src/aws/LabelsRekognition.py
import boto3
from botocore.exceptions import BotoCoreError, ClientError
from config import config

# Crear cliente de Rekognition
rekog_client = boto3.client(
    'rekognition',
    region_name=config['rekognition']['region'],
    aws_access_key_id=config['rekognition']['accessKeyId'],
    aws_secret_access_key=config['rekognition']['secretAccessKey']
)

bucket = config['bucket_s3']

def detect_labels(photo):
    try:
        response = rekog_client.detect_labels(
            Image={
                'S3Object': {
                    'Bucket': bucket,
                    'Name': photo
                }
            }
        )
        etiquetas = []
        for label in response['Labels']:
            etiquetas.append(label['Name'])
            # Puedes descomentar el código siguiente para más detalles
            # print(f"Confidence: {label['Confidence']}")
            # print(f"Name: {label['Name']}")
            # print("Instances:")
            # for instance in label.get('Instances', []):
            #     print(instance)
            # print("Parents:")
            # for parent in label.get('Parents', []):
            #     print(parent)
            # print("-------")
        return etiquetas  # Para pruebas unitarias
    except (BotoCoreError, ClientError) as err:
        print("Error", err)
        return None

rekognition = {
    "detect_labels": detect_labels
}